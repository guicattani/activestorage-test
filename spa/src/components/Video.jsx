import React from "react";

import "./Video.css";

const Video = (props) => {
  const { url, title, category, thumbnail } = props;

  return (
    <div className="card video" data-testid="video-card">
      <video
        controls
        src={url}
        preload="none"
        poster={thumbnail}
        title={title}
        className="video-player"
      />

      <div className="font-bold text-xl mb-2">{title}</div>
      <span className="float-right inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2">
        {category}
      </span>
    </div>
  );
};

export default Video;
