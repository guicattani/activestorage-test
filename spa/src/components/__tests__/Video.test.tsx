import React  from "react"
import { render, screen, cleanup } from "@testing-library/react"
import renderer from 'react-test-renderer'
import Video from "../Video"

afterEach(()=> {
    cleanup();
})

const videoTestProps = { url:  'https://www.caspar-health.com', title: "shoulder exercise" }


test('should render video component', ()=> {
    render(<Video video= {videoTestProps}/>)  
}) 


test('matches the snapshot', ()=> {
    const tree = renderer.create(<Video video={videoTestProps} />)
    expect(tree).toMatchSnapshot();
})