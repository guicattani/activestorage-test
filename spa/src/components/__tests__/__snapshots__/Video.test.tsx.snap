// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="card video"
  data-testid="video-card"
>
  <video
    className="video-player"
    controls={true}
    preload="none"
  />
  <div
    className="font-bold text-xl mb-2"
  />
  <span
    className="float-right inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
  />
</div>
`;
